lib/matplotlib/cbook.py

```diff
@@ -2487,6 +2490,41 @@ def pts_to_midstep(x, *args):
                    'step-post': pts_to_poststep,
                    'step-mid': pts_to_midstep}
 
+
+def get_index_y(y):
+    """
+    A helper function to get the index of an input to plot
+    against if x values are not explicitly given.
+
+    Tries to get `y.index` (works if this is a pd.Series), if that
+    fails, return np.arange(y.shape[0]).
+
+    This will be extended in the future to deal with more types of
+    labeled data.
+
+    Parameters
+    ----------
+    y : scalar or array-like
+        The proposed y-value
+
+    Returns
+    -------
+    x, y : ndarray
+       The x and y values to plot.
+    """
+    try:
+        return y.index.values, y.values
+    except AttributeError:
+        y = np.atleast_1d(y)
+        return np.arange(y.shape[0], dtype=float), y
+
+
+def get_label(y, default_name):
```

Comment:
Extend it how?  This exists as just a plain function, so we can't extend it in the class sense... if we look for objects storing the label in some other way, then we need a specific implementation for that and not a generic cook-book one, right?